### PR TITLE
Fakes fixes

### PR DESCRIPTION
**Pull request by tommy9**
Fixes https://github.com/rubberduck-vba/Rubberduck/issues/4476, fixes https://github.com/rubberduck-vba/Rubberduck/issues/5944

The fix for 4476 is simply to change the list of return values from being static as that meant it was shared across multiple fakes and so a return value intended for one fake could be used by another, causing type conversion errors (as well as unintended values).

The fix for 5944 introduces a way to disable another fake temporarily. This is necessary when one hooked function uses the passthrough feature to call the native function which itself contains a call to another hooked function. We don't want that behaviour anyway as someone hooking the Now function for example would not expect it to change the result of the Date function. I've done this for the Date and Time fakes to avoid issues with the Now fake. I had a quick look through other functions and didn't spot other potential issues but if they do arise the same approach should be useful.

The thing I'm not so keen on in the solution is using a string comparison of the types. I tried to use type comparisons directly but couldn't get it to work. My c# level is pretty basic. Happy to get any pointers to tidy this up.

### CODE REVIEW

File: Rubberduck.Main/ComClientLibrary/UnitTesting/FakesProvider.cs [hunk 1]
```diff
@@ -37,6 +37,30 @@ public void StartTest()
...
+            }
+        }
+
+        public static void ResumeFake(String typename)
+        {
+            foreach (var fake in ActiveFakes.Values)
+            {
+                if (fake.GetType().Name == typename)
+                {
+                    fake.DisableHook();
```
Comment (retailcoder) [inline Rubberduck.Main/ComClientLibrary/UnitTesting/FakesProvider.cs:58]: The implementation is identical to that of `SuspendFake`, is that intended?

File: Rubberduck.Main/ComClientLibrary/UnitTesting/FakesProvider.cs [hunk 2]
```diff
@@ -37,6 +37,30 @@ public void StartTest()
             CodeIsUnderTest = true;
         }
 
+        public static void SuspendFake(String typename)
```
Comment (retailcoder) [inline Rubberduck.Main/ComClientLibrary/UnitTesting/FakesProvider.cs:40]: Could take a `Type` parameter here

File: Rubberduck.Main/ComClientLibrary/UnitTesting/FakesProvider.cs [hunk 3]
```diff
@@ -37,6 +37,30 @@ public void StartTest()
...
+        {
+            foreach (var fake in ActiveFakes.Values)
+            {
+                if (fake.GetType().Name == typename)
```
Comment (retailcoder) [inline Rubberduck.Main/ComClientLibrary/UnitTesting/FakesProvider.cs:44]: ...then this becomes `if (fake.GetType() == type)`, no?

**Review by retailcoder — CHANGES_REQUESTED** [review]
Thank you! tbh the stringly-typed check isn't a showstopper at all; if the strongly-typed check can't work then changing the hard-coded strings to e.g. `typeof(Now).Name` should be more refactoring-friendly.

File: Rubberduck.Main/ComClientLibrary/UnitTesting/FakesProvider.cs (hunk 1 above)
Comment (tommy9) [reply Rubberduck.Main/ComClientLibrary/UnitTesting/FakesProvider.cs:58]: Definitely wasn't intended! I'll fix and also enhance the test case to cover the reenabling of the hook.

File: Rubberduck.Main/ComClientLibrary/UnitTesting/FakesProvider.cs (hunk 3 above)
Comment (tommy9) [reply Rubberduck.Main/ComClientLibrary/UnitTesting/FakesProvider.cs]: Perfect, thanks!

**Review by retailcoder — APPROVED** [review]

**Pull request merged**